cmd/kube-scheduler/app/server.go

```diff
@@ -307,12 +310,22 @@ func Run(ctx context.Context, cc *schedulerserverconfig.CompletedConfig, sched *
 
 		leaderElector.Run(ctx)
 
+		// for graceful shutdown
+		<-serverShutdownCh
+		<-listenerStoppedCh
```

Comment:
Are we sure `cc.SecureServing != nil` here?

Reply:
I addressed your advice by using the defer statement.
[use defer](https://github.com/kubernetes/kubernetes/pull/128888/commits/9cc325e79d67c64132431b93816a4b5d92c4a7d3)